packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -0,0 +1,1459 @@
+    }
+
+    /**
+     * @notice Constructs the delegator of KGH in the vault of a validator.
+     *
+     * @custom:field kroShares              Amount of shares for KRO delegation.
+     * @custom:field kghShares              Amount of shares for delegation of KGH with a corresponding tokenId.
+     * @custom:field undelegateRequestTimes Timestamps of undelegation requests.
+     * @custom:field undelegateRequestedIds A mapping of timestamp undelegations are requested to
+     *                                      tokenIds of KGHs for undelegation.
+     * @custom:field undelegateRequests     A mapping of timestamp undelegations are requested to
+     *                                      pending KGH shares for undelegation.
+     */
+    struct KghDelegator {
+        mapping(uint256 => uint128) kroShares;
+        mapping(uint256 => uint128) kghShares;
+        uint256[] undelegateRequestTimes;
+        mapping(uint256 => uint256[]) undelegateRequestedIds;
+        mapping(uint256 => uint128) undelegateRequests;
+    }
+
+    /**
+     * @notice Constructs the vault of a validator.
+     *
+     * @custom:field asset             Asset information of the vault.
+     * @custom:field reward            Reward information of the vault.
+     * @custom:field isInitiated       Whether the vault is initiated.
+     * @custom:field noSubmissionCount Number of counts that the validator did not submit the output in priority round.
+     * @custom:field kroDelegators     A mapping of validator address to KRO delegator struct.
+     * @custom:field kghDelegators     A mapping of validator address to KGH delegator struct.
+     */
+    struct Vault {
+        Asset asset;
+        Reward reward;
+        bool isInitiated;
+        uint8 noSubmissionCount;
+        mapping(address => KroDelegator) kroDelegators;
+        mapping(address => KghDelegator) kghDelegators;
+    }
+
+    /**
+     * @notice The numerator for the boosted reward.
+     */
+    uint128 public constant BOOSTED_REWARD_NUMERATOR = 40;
+
+    /**
+     * @notice The denominator for the boosted reward.
+     */
+    uint128 public constant BOOSTED_REWARD_DENOM = 100;
+
+    /**
+     * @notice The denominator for per-mille used at slashing and initialization of undelegation.
+     */
+    uint128 public constant PER_MILLE_DENOM = 1000;
+
+    /**
+     * @notice Virtual KRO amount per KGH.
+     */
+    uint128 public constant VKRO_PER_KGH = 100e18;
+
+    /**
+     * @notice The numerator of the tax.
+     */
+    uint128 public constant TAX_NUMERATOR = 20;
+
+    /**
+     * @notice The denominator of the tax.
+     */
+    uint128 public constant TAX_DENOMINATOR = 100;
+
+    /**
+     * @notice Decimals offset for the KRO and KGH shares.
+     */
+    uint128 public constant DECIMAL_OFFSET = 10 ** 6;
+
+    /**
+     * @notice The maximum number of KROs that can be undelegated at once.
+     */
+    uint256 public constant MAX_DELEGATE = 50_000_000 * 10 ** 18;
```

Comment:
That was a convention of ERC-4626, and it was intended to prevent the cases that `receiver` (deposit to another account is allowed at ERC-4626) takes more than the max amount he can receive.

I fully agree that it is useless in this case, so I'll remove it. Thanks!